src/main/java/com/epam/jwd/task01/logic/validator/GemstoneValidator.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.jwd.task01.logic.validator;
+
+import com.epam.jwd.task01.entity.jewel.Gemstone;
+
+import java.math.BigDecimal;
+
+/**
+ * Method for validation of some fields values, specified at the time of creation of <tt>new</tt>
+ * instances of {@link Gemstone}.
+ * <p>Throws <tt>IllegalArgumentException</tt> if specified values are out of a given range.
+ */
+public class GemstoneValidator {
```

Comment:
You may need other validators in the future, so it's better to create an interface Validator with a method validate()